### PR TITLE
feat: refactor config and add ChainConfig

### LINKED ISSUE #106: Sidecar: refactor the `config` module
The config.rs file is getting crowded already. 
It's a good time to turn it into a directory with:

- mod.rs: same content as current `config.rs`
- chain.rs: Chain configs only

_Originally posted by @merklefruit in https://github.com/chainbound/bolt/pull/104#discussion_r1664115958_

### PR DESCRIPTION
**Pull request by merklefruit**
- fixes #106 

## Summary by CodeRabbit

- **New Features**
  - Introduced `ChainConfig` for enhanced chain configuration management, providing defaults for commitment deadlines and slot times.

- **Refactor**
  - Renamed `Chain` to `ChainConfig` across various modules for better clarity and organization.
  - Updated configuration and signature methods to utilize `ChainConfig`.
  - Reorganized configuration modules to better structure chain and signing options.

- **Chores**
  - Removed outdated constants and fields related to `commitment_deadline`.
  - Enhanced command-line options with new `SigningOpts` struct for signing configurations.

### CODE REVIEW

File: bolt-sidecar/src/config/chain.rs
Comment (merklefruit) [inline bolt-sidecar/src/config/chain.rs:1]: this is made to work like before: we have default values for all chains, and CLI arguments "slot_time_in_seconds" and "commitment_deadline" are optional and override the defaults.

File: bolt-sidecar/src/config/chain.rs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    #[clap(short = 'C', long, default_value = "mainnet")]
+    chain: Chain,
+    /// The deadline in the slot at which the sidecar will stop accepting
+    /// new commitments for the next block (parsed as milliseconds).
+    #[clap(short = 'd', long, default_value_t = DEFAULT_COMMITMENT_DEADLINE_IN_MILLIS)]
+    commitment_deadline: u64,
+    /// The slot time duration in seconds. If provided,
+    /// it overrides the default for the selected [Chain].
+    #[clap(short = 's', long, default_value_t = DEFAULT_SLOT_TIME_IN_SECONDS)]
+    slot_time_in_seconds: u64,
```
Comment (mempirate) [inline bolt-sidecar/src/config/chain.rs:46]: Let's just do `slot_time` and add a description saying it's in seconds!

File: bolt-sidecar/src/config/signing.rs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#[clap(
+    group = ArgGroup::new("signing-opts").required(true)
+        .args(&["private_key", "commit_boost_url"])
+)]
+pub struct SigningOpts {
+    /// Private key to use for signing preconfirmation requests
+    #[clap(short = 'k', long)]
+    pub(super) private_key: Option<String>,
+    /// URL for the commit-boost sidecar
+    #[clap(short = 'C', long, conflicts_with("private_key"))]
```
Comment (mempirate) [inline bolt-sidecar/src/config/signing.rs:14]: Didn't know `conflicts_with`, sick!

**Comment by mempirate** [comment]
@coderabbitai review

File: bolt-sidecar/src/config/signing.rs (hunk 2 above)
Comment (merklefruit) [reply bolt-sidecar/src/config/signing.rs:14]: just ported over, work of @thedevbirb :)

**Review by namn-grg — COMMENTED** [review]
LGTM, good refactor!

**Pull request merged**